File: practice/L33t/867transposeMatrix.js
/**
 * Given a matrix A, return the transpose of A.
 * The transpose of a matrix is the matrix flipped over it's main diagonal, switching the row and column indices of the matrix.
 * @param {number[][]} A
 * @return {number[][]}
 */
var transpose = function(A) {
    let newMatrix = new Array()
    for(let i=0;i<A[0].length;i++){
        let temp = new Array()
        for(let j=0;j<A.length;j++){
            temp.push(A[j][i])
        }
        newMatrix.push(temp)
    }
    return newMatrix
};

File: challenges/1337/867transposeMatrix.swift
// Given a 2D integer array matrix, return the transpose of matrix.
// The transpose of a matrix is the matrix flipped over its main diagonal, switching the matrix's row and column indices.

class Solution {
    func transpose(_ matrix: [[Int]]) -> [[Int]] {
        var sol = Array(repeating: Array(repeating: 0, count: matrix.count), count: matrix[0].count)
        for i in 0..<matrix.count {
            for j in 0..<matrix[0].count {
                sol[j][i] = matrix[i][j]
            }
        }
        return sol
    }
}